scipy/stats/_stats_py.py

```diff
@@ -428,16 +430,17 @@ def pmean(a, p, *, axis=0, dtype=None, weights=None):
     if p == 0:
         return gmean(a, axis=axis, dtype=dtype, weights=weights)
 
-    a = np.asarray(a, dtype=dtype)
+    xp = array_namespace(a, weights)
+    a = xp.asarray(a, dtype=dtype)
 
     if weights is not None:
-        weights = np.asanyarray(weights, dtype=dtype)
+        weights = xp.asarray(weights, dtype=dtype)
 
     negative_mask = a < 0
-    if np.any(negative_mask):
+    if xp.any(negative_mask):
         # `where` avoids having to be careful about dtypes and will work with
         # JAX. This is the exceptional case, so it's OK to be a little slower.
-        a = np.where(negative_mask, np.nan, a)
+        a = xp.where(negative_mask, np.nan, a)
```

Comment:
nit but I assume the same comment from above applies?
```suggestion
        # Won't work for array_api_strict for now, but see data-apis/array-api#807
        a = xp.where(negative_mask, np.nan, a)
```